go/parquet/internal/debug/log_off.go

```diff
@@ -0,0 +1,21 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !debug
+
+package debug
+
+func Log(interface{}) {}
```

Comment:
comment? Is this a logging interface?  does Go have a common library for this?

Reply:
this is using the build tags to have a debug logging interface that can be enabled by passing `-tags debug` when building so that they get optimized away in a build without that flag. It's the equivalent of using macros in C++ to check `#ifdef NDEBUG` etc. The go arrow library does the same.

Go has a common library for logging, but it's designed to be used in conjunction with env vars controlling log level output rather than using the build tags to control optimizing away the debug logs like this.